### PR TITLE
ovirt add host remove retry

### PR DESCRIPTION
**Pull request by mnecas**
##### SUMMARY

ovirt add host remove retry

Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1719271
##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

ovirt
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: lib/ansible/modules/cloud/ovirt/ovirt_host.py [hunk 1]
```diff
@@ -367,6 +367,16 @@ def failed_state_after_reinstall(self, host, count=0):
         return False
 
 
+def remove_host(host_module, host, count=0):
+    try:
+        return host_module.remove()
+    except Exception as e:
+        if count > 5:
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_host.py]: Please make the sleep and retry limits configurable by parameters.

Comment (machacekondra) [reply lib/ansible/modules/cloud/ovirt/ovirt_host.py]: Also using recursin isn't really needed here, please use normal iteration.

File: lib/ansible/modules/cloud/ovirt/ovirt_host.py [hunk 2]
```diff
@@ -367,6 +367,16 @@ def failed_state_after_reinstall(self, host, count=0):
         return False
 
 
+def remove_host(host_module, host, max_count=5, timeout=20):
+    for count in range(max_count + 1):
+        try:
+            return host_module.remove()
+        except Exception as e:
+            if count == max_count:
```
Comment (machacekondra) [inline lib/ansible/modules/cloud/ovirt/ovirt_host.py]: Please pass wait module parameter, and do no try to retry the removal in case wait is false.

**Review by resmo — COMMENTED** [review]
I am skeptical about this because the new args seems to be related for a workaround of an issue of ovirt. 

I would avoid adding new args just for a workaround. 

Further, the workaround could even be implemented with default ansible functionality "until" https://docs.ansible.com/ansible/latest/user_guide/playbooks_loops.html#retrying-a-task-until-a-condition-is-met, so probably a "update the docs" would just be enough?

**Pull request closed without merging**